test/integration/client/exec_test.go

```diff
@@ -0,0 +1,329 @@
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s",
+								"clientCertificateData": %s,
+								"clientKeyData": %s
+							}
+						}`, clientUnauthorizedToken, read(t, clientCertFileName), read(t, clientKeyFileName)),
+				})
+			},
+		},
+		{
+			name: "authorized token and unauthorized certificate favors authorized token",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s",
+								"clientCertificateData": %q,
+								"clientKeyData": %q
+							}
+						}`, clientAuthorizedToken, string(unauthorizedCert), string(unauthorizedKey)),
+				})
+			},
+		},
+		{
+			name: "unauthorized token and unauthorized certificate",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s",
+								"clientCertificateData": %q,
+								"clientKeyData": %q
+							}
+						}`, clientUnauthorizedToken, string(unauthorizedCert), string(unauthorizedKey)),
+				})
+			},
+			wantClientErrorPrefix: "Unauthorized",
+		},
+		{
+			name: "good token with static auth basic creds favors static auth basic creds",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s"
+							}
+						}`, clientAuthorizedToken),
+				})
+				c.Username = "unauthorized"
+				c.Password = "unauthorized"
+			},
+			wantClientErrorPrefix: "Unauthorized",
+		},
+		{
+			name: "good token with static auth bearer token favors static auth bearer token",
+			clientConfigFunc: func(c *rest.Config) {
+				c.ExecProvider.Env = append(c.ExecProvider.Env, clientcmdapi.ExecEnvVar{
+					Name: outputEnvVar,
+					Value: fmt.Sprintf(`{
+							"kind": "ExecCredential",
+							"apiVersion": "client.authentication.k8s.io/v1beta1",
+							"status": {
+								"token": "%s"
+							}
+						}`, clientAuthorizedToken),
+				})
+				c.BearerToken = "some-unauthorized-token"
+			},
+			wantClientErrorPrefix: "Unauthorized",
+		},
+		{
+			name: "good token with static auth cert and key favors exec plugin",
```

Comment:
uh is this what we want to happen?

Reply:
hmmm. i think i had convinced myself that this is what we wanted to happen because the exec authenticator doesn't have any way of detecting that the underlying transport is using a static auth cert and key. altho, that is merely an implementation argument, which might not capture what the user would expect. were you thinking that the user would expect that the static auth cert and key would completely supersede any authentication material returned by the exec authenticator?

Reply:
> were you thinking that the user would expect that the static auth cert and key would completely supersede any authentication material returned by the exec authenticator?

Isn't a similar assertion made by the `good token with static auth basic creds favors static auth basic creds` test case?

Reply:
5afdaa85ed57533ef46af75c55d8c87f1ced2b76